qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -148,6 +151,129 @@ def model_names(self) -> List[str]:
+                    "Residuals plotting is currently supported for analysis with 1 subplot.",
+                    UserWarning,
+                    stacklevel=2,
+                )
+                fields["plot_residuals"] = False
+            else:
+                self._add_residuals_plot_config()
+        if not fields.get("plot_residuals", True) and self.options.get("plot_residuals"):
+            self._remove_residuals_plot_config()
+
+        super().set_options(**fields)
+
+    def _add_residuals_plot_config(self):
+        """Configure plotter options for residuals plot."""
+        # check we have model to fit into
+        if self.models:
+            # Cache figure options.
+            self._plot_config_cache["figure_options"] = {}
+            self._plot_config_cache["figure_options"]["ylabel"] = self.plotter.figure_options.get(
+                "ylabel"
+            )
+            self._plot_config_cache["figure_options"][
+                "series_params"
+            ] = self.plotter.figure_options["series_params"]
+            self._plot_config_cache["figure_options"]["sharey"] = self.plotter.figure_options[
+                "sharey"
+            ]
+
+            self.plotter.set_figure_options(
+                ylabel=[
+                    self.plotter.figure_options.get("ylabel", ""),
+                    "Residuals",
+                ],
+            )
+            model_names = self.model_names()
+            for model_name in model_names:
+                self.plotter.set_figure_options(
+                    sharey=False,
+                    series_params={
+                        **self.plotter.figure_options["series_params"],
+                        **{
+                            model_name: {
+                                **{
+                                    "canvas": 0,
+                                },
+                                **self.plotter.figure_options["series_params"].get(model_name, {}),
+                            },
+                            model_name
+                            + "_residuals": {
+                                **{
+                                    "canvas": 1,
+                                },
+                                **self.plotter.figure_options["series_params"].get(model_name, {}),
+                            },
+                        },
+                    },
+                )
+
+            # Cache plotter options.
+            self._plot_config_cache["plotter"] = {}
+            self._plot_config_cache["plotter"]["subplots"] = self.plotter.options.get("subplots")
+            self._plot_config_cache["plotter"]["style"] = self.plotter.options.get(
+                "style", PlotStyle({})
+            )
+
+            # removing the name from the plotter style, so it will not clash with the new name
+            previous_plotter_style = self._plot_config_cache["plotter"]["style"].copy()
+            previous_plotter_style.pop("style_name", "")
```

Comment:
What's the name that's being removed here? Is it something that's displayed in the figure?

Reply:
I want to merge both styles of the figure but I want to keep the `style_name` unique so I remove the `style_name` (if exist) from the `PlotStyle` before trying to merge.